tests/unittests/Foundation/NSDecimalNumberTests.mm

```diff
@@ -0,0 +1,275 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#import <TestFramework.h>
+#import <Foundation/Foundation.h>
+
+#define ASSERT_ARRAY_EQUAL(a, b)                     \
+    do {                                             \
+        for (int i = 0; i < NSDecimalMaxSize; ++i) { \
+            ASSERT_EQ(a[i], b[i]);                   \
+        }                                            \
+    } while (0, 0)
+
+static void testSubtract(NSDecimalNumber* x, NSDecimalNumber* y, NSDecimalNumber* expected) {
+    NSDecimalNumber* result = [x decimalNumberBySubtracting:y];
+    EXPECT_OBJCEQ(expected, result);
+}
+
+static void testAddition(NSDecimalNumber* x, NSDecimalNumber* y, NSDecimalNumber* expected) {
+    NSDecimalNumber* result = [x decimalNumberByAdding:y];
+    EXPECT_OBJCEQ(expected, result);
+}
+
+TEST(NSDecimalNumber, InitWithLongLong) {
+    unsigned short mantissa[] = { 21495, 42467, 50331, 32, 0, 0, 0, 0, 0 };
+    NSDecimalNumber* dNumber = [[[NSDecimalNumber alloc] initWithLongLong:9223372036854775] autorelease];
+
+    NSDecimal decimal = [dNumber decimalValue];
+    ASSERT_EQ(4, decimal._length);
+    ASSERT_EQ(0, decimal._exponent);
+    ASSERT_EQ(0, decimal._isNegative);
+
+    ASSERT_ARRAY_EQUAL(mantissa, decimal._mantissa);
+
+    dNumber = [[[NSDecimalNumber alloc] initWithLongLong:-9223372036854775] autorelease];
```

Comment:
Yep Limits tests are being added

---
In reply to: [80506974](https://github.com/Microsoft/WinObjC/pull/1032#discussion_r80506974) [](ancestors = 80506974)